### ISSUE TITLE
Temporary UI freeze when receiving feed

### ISSUE THREAD

**Issue by nottelmo** [jangernert/FeedReader#773]
Using Fedora 28, Gnome 3.28

Everytime FeedReader fetches new News, I get a temporary UI freeze

**Comment by bilelmoussaoui** [comment]
Which version are you using? The latest Flatpak one?

**Comment by nottelmo** [comment]
@bilelmoussaoui Yes. And i have to correct myself that it doesn't happen everytime. I have been unable to verify when that happens 100% of the time, but I think I am pretty sure that it freezes when i'm watching a video on Gnome Videos and it fetches latest news.

**Comment by jangernert** [comment]
I know this can happen when a lot of articles are visible (even when they are above or below the article list viewport) in the list. And possibly also when the feed list is quite long (even if most of them are collapsed). FeedReader isn't too smart about updating the UI and sorta goes through everything and updated widget by widget.
I am working on something better for the Rust rewrite. But progress strongly varies depending on real life stuff.
